src/main/java/com/datalogics/pdf/hsm/samples/HsmSignDocument.java

```diff
@@ -70,30 +77,34 @@ public static void main(final String... args) throws Exception {
         // If you are not using an evaluation version of the product you can ignore or remove this code.
         LicenseManager.setLicensePath(".");
 
+        hsmManager = HsmManagerFactory.newInstance(HsmManagerFactory.LUNA_HSM_TYPE);
+
         URL outputUrl = null;
         if (args.length > 0) {
             outputUrl = new File(args[0]).toURI().toURL();
         } else {
             outputUrl = new File(OUTPUT_SIGNED_PDF_PATH).toURI().toURL();
         }
 
-        final URL inputUrl = HsmSignDocument.class.getResource(INPUT_UNSIGNED_PDF_PATH);
-
-        // Log in to the HSM
-        final boolean loggedIn = hsmManager.hsmLogin(TOKEN_LABEL, PASSWORD);
+        if (!hsmManager.isLoggedIn()) {
+            // Log in to the HSM
+            hsmManager.hsmLogin(new LunaHsmLoginParms(TOKEN_LABEL, PASSWORD));
+        }
```

Comment:
I'm doubtful about doing a check if the `hsmManager` is logged in here: since we're in `main()`, we've just procured it, and it's not a singleton, it's guaranteed at this point NOT to be logged in. It certainly doesn't hurt to do it here, but it seems redundant.

Do @jrowden8 or @datalogics-kam have a strong argument for why it SHOULD be checked first?